### DISCUSSION TITLE
Youtube closed captions not showing up in smplayer

### DISCUSSION THREAD

**Discussion by tack32** [smplayer-dev/smplayer#654] [General]
I had it working earlier on another video from Youtube but this Youtube video has closed captions when viewed on Youtube but I can't get them on smplayer.https://postimg.cc/nMJFN0mz I have it set to yt-dlp. On the one that worked in subtitles/secondary it showed https://postimg.cc/94Pmf95V 
However, now, on the one that worked before, I no longer see the subtitles there either. Not sure if I did something different in settings? Thanks

**Comment by smplayer-dev** [comment]
In order to get subtitles from youtube you need to set the option `support for video sites` to `mpv + youtube_dl` in preferences -> network.

**Reply by tack32** [reply]
Appreciate your reply. It seems to me I already did that? This is what I see in my settings. By the way the Youtube video is https://www.youtube.com/watch?v=6oRhdu4ax5I
![subtitles settings but no go cotton bowl](https://user-images.githubusercontent.com/93066797/220201200-1a42a2cb-be52-4876-8d40-be0bb9ea8914.jpg)
Thanks

**Reply by tack32** [reply]
It's working now. Maybe there was something about that particular video. The next one I tried https://www.youtube.com/watch?v=4yjRZF5WuPo is showing the subtitles or closed captions in smplayer. Thanks

**Reply by n3thshan** [reply]
> In order to get subtitles from youtube you need to set the option support for video sites to mpv + youtube_dl in preferences -> network.

This option is not available in the flatpak version of smplayer. if so how can i get captions to work?

![image](https://github.com/user-attachments/assets/32f53868-a30e-4e40-9f77-999def994383)

**Comment by smplayer-dev** [comment]
The option "mpv + youtube_dl" was renamed to just "mpv".

**Comment by n3thshan** [comment]
does this work for automatic captions as well? (does not work for me) how do i properly use it if so? (i.e. which of 4 closed captions do i choose and what do they mean?) 

EDIT: nevermind, i created a config file next to the yt-dlp binary with the following arguments and it works now. 

```
--write-auto-subs
--sub-langs en
--embed-subs
```

**Reply by n3thshan** [reply]
unfortunately, however, i cannot seem to get smplayer to display english subtitles only. auto subs in all languages appear. is there a way to fix it?
